File: app/src/main/java/com/tanacom/introrealm/java/CountryAdapter.java
//package com.tanacom.introrealm.java;
//
//import android.view.LayoutInflater;
//import android.view.View;
//import android.view.ViewGroup;
//import android.widget.TextView;
//
//import androidx.annotation.NonNull;
//import androidx.annotation.Nullable;
//import androidx.recyclerview.widget.RecyclerView;
//
//import com.tanacom.introrealm.R;
//import com.tanacom.introrealm.kotlin.CountryModel;
//
//import io.realm.OrderedRealmCollection;
//import io.realm.RealmRecyclerViewAdapter;
//
//public class CountryAdapter extends RealmRecyclerViewAdapter<CountryModel, CountryAdapter.MyViewHolder> {
//
//    public CountryAdapter(@Nullable OrderedRealmCollection<CountryModel> data) {
//        super(data, true);
//
//        // Used when there's primary key in Model
//        setHasStableIds(true);
//    }
//
//    @NonNull
//    @Override
//    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
//        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_country, parent, false);
//        return new MyViewHolder(itemView);
//    }
//
//    @Override
//    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
//
//        final CountryModel model = getItem(position);
//
//        holder.tvCountry.setText(model.getName());
//
//    }
//
//    @Override
//    public long getItemId(int position) {
//        return getItem(position).getId();
//    }
//
//    static class MyViewHolder extends RecyclerView.ViewHolder {
//
//        TextView tvCountry;
//
//        public MyViewHolder(@NonNull View itemView) {
//            super(itemView);
//
//            tvCountry = itemView.findViewById(R.id.tv_item);
//        }
//    }
//
//
//}

File: app/src/main/java/com/tanacom/introrealm/kotlin/CountryAdapter.kt
//package com.tanacom.introrealm.kotlin
//
//import android.view.LayoutInflater
//import android.view.View
//import android.view.ViewGroup
//import android.widget.TextView
//import androidx.recyclerview.widget.RecyclerView
//import com.tanacom.introrealm.R
//import io.realm.OrderedRealmCollection
//import io.realm.RealmRecyclerViewAdapter
//
//class CountryAdapter(data: OrderedRealmCollection<CountryModel?>?) :
//
//    RealmRecyclerViewAdapter<CountryModel?, CountryAdapter.MyViewHolder>(data, true) {
//
//    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
//        val itemView =
//            LayoutInflater.from(parent.context).inflate(R.layout.item_country, parent, false)
//        return MyViewHolder(itemView)
//    }
//
//    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
//        val model = getItem(position)
//        holder.tvCountry.text = model!!.name
//    }
//
//    override fun getItemId(position: Int): Long {
//        return getItem(position)!!.id.toLong()
//    }
//
//    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
//
//        var tvCountry: TextView = itemView.findViewById(R.id.tv_item)
//
//    }
//
//    init {
//        // Used when there's primary key in Model
//        setHasStableIds(true)
//    }
//
//
//}